tools/build-image.sh

```diff
@@ -188,11 +188,27 @@ cp ${imgdir}/microOS/config.{sh,xml} ${build}/
 bundle || exit 1
 
 mkdir ${build}/{_out,_logs}
-(set -o pipefail
-sudo kiwi-ng --debug --profile=${profile} --type oem \
-  system build --description ${build} \
-  --target-dir ${build}/_out |\
-  tee ${build}/_logs/${build_name}-build.log)
 
-exit $?
+osid=$(grep '^ID=' /etc/os-release | sed -e 's/\(ID=["]*\)\(.\+\)/\2/' | tr -d '"')
+case $osid in
+  opensuse-tumbleweed | opensuse-leap)
+    (set -o pipefail
+    sudo kiwi-ng --debug --profile=${profile} --type oem \
+      system build --description ${build} \
+      --target-dir ${build}/_out |\
+      tee ${build}/_logs/${build_name}-build.log)
+    ;;
+  debian | ubuntu)
+    (set -o pipefail
+    sudo kiwi-ng --debug --profile=${profile} --type oem \
+      system boxbuild --box tumbleweed --no-update-check -- --description ${build} \
+      --target-dir ${build}/_out |\
+      tee ${build}/_logs/${build_name}-build.log)
+    ;;
+  *)
+    echo "error: unsupported distribution ($osid) kiwi-ng may not work"
+    exit 1
+      ;;
+esac
 
+exit $?
```

Comment:
I wonder though whether this return code is going to be for the actual command that was run, or, as bash sometimes likes to do, the last statement (which I'm guessing was a successful `case`).

As much as I hate to hold up the PR yet again, mind adding the `exit $?` right after the calls to kiwi (mimicking what had been done)? Unless you are certain this is propagating the error as expected?

Reply:
sound good, I just move the exit $? into the case switch. 
Meanwhile I updated btrfs file check with different distro location